### PR TITLE
Add REST API for alerts management

### PR DESCRIPTION
**Pull request by menakaj**
This PR introduces alert management api for Publisher and Store.

- Get the supported alert types.
- Subscribe to the alert types.
- Add configuration to the alerts.
- Unsubscribe from alerts.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    curl: |
+        curl -k -H "Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8" https://localhost:9443/api/am/publisher/v1.0/alert-types
+    rawResponse: |
+      {
+        "count":2,
+        "alerts":[
+          {
+            "id":3,
+            "name":"AbnormalRequestTime",
+            "isRequireConfiguration":true
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts.yaml]: shall we make it `requireConfiguration`?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+          "configurationId":"UGl6emFTaGFja0FQSSMxLjAuMA",
+          "configuration":
+            {
+              "requestTime":"120",
+              "apiName":"PizzaShackAPI",
+              "apiVersion":" 2.0",
+            }
+        }
+
+  - rawRequest: |
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml]: We can use 'rawRequest' only when the requests and responses are not easily structured to headers, parsable body etc. We do not need to set the same example in both formats. Shall we remove this?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      headers: |
+        Authorization: Bearer eff01a97-4dc0-3542-a184-75aecf355ec2
+        Content-Type: application/json
+      response:
+        status:
+          code: 200
+          msg: OK
+        headers: |
+          Content-Type: application/json
+  - rawRequest: |
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml]: Let's remove the 'rawRequest'. No need to give the example in both formats.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml [hunk 4]
```diff
@@ -8753,6 +9058,141 @@ definitions:
...
+    properties:
+      id:
+        type: integer
+        description: The alert Id
+        example: 1
+      name:
+        type: string
+        description: The name of the alert.
+        example: "AbnormalRequestsPerMin"
+      isRequireConfiguration:
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Shall we rename this to requireConfiguration: true/false

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/docs/examples/alerts/alerts.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            "name":"AbnormalRequestsPerMin",
+            "isRequireConfiguration":true
+          },
+          {
+            "id":4,
+            "name":"AbnormalRequestPattern",
+            "isRequireConfiguration":false
+          }]
+        }
+  - rawRequest: |
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/docs/examples/alerts/alerts.yaml]: Let's remove these as mentioned above.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts.yaml (hunk 1 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml (hunk 2 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml (hunk 3 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 4 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/docs/examples/alerts/alerts.yaml (hunk 5 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/docs/examples/alerts/alerts.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts.yaml (hunk 1 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml (hunk 2 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml (hunk 3 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/docs/examples/alerts/alerts_config.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml (hunk 4 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/resources/publisher-api.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/docs/examples/alerts/alerts.yaml (hunk 5 above)
Comment (menakaj) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/docs/examples/alerts/alerts.yaml]: Fixed in https://github.com/wso2/carbon-apimgt/pull/7117

**Pull request merged**